### PR TITLE
Better es6 class support

### PR DESCRIPTION
**Pull request by dobesv**
Use Reflect to construct class in a more compatible manner.

Reflect should be available in environments that use es6 classes.

This can fix https://github.com/FormidableLabs/radium/issues/999 because we preserve the same value of `this` as was used in the object constructor, whereas the previous implementation discarded that object and used a new object after calling the constructor.  That resulted in some confusing behavior if the constructor stored referenced to `this` or bound methods / functions to `this`.

### CODE REVIEW

**Comment by ryan-roemer** [comment]
Thanks for opening this!!!

In case it helps as we're getting ci fixed up on this, probably about time to switch Travis versions to:

```yml
node_js:
  - "6"
  - "8"
  - "10"
```

(Appveyor is already there...)

**Comment by ryan-roemer** [comment]
Something we've added is failing on uglify

```
ERROR in radium.min.js from UglifyJs
Unexpected token: punc (.) [radium.min.js:508,11]
[builder:proc:end:2] Command: webpack --config=webpack.config.minified.js
[builder:proc:error] Code: 2, Command: webpack --config=webpack.config.minified.js
[builder:builder-core:end:5230] Task: concurrent build-dist-dev, build-dist-prod, Error: Command failed: sh -c webpack --config=webpack.config.minified.js
[builder:proc:end:2] Command: builder concurrent --buffer build-dist-dev build-dist-prod
[builder:proc:error] Code: 2, Command: builder concurrent --buffer build-dist-dev build-dist-prod
[builder:builder-core:end:5215] Task: concurrent build-lib, build-dist, build-es, Error: Command failed: sh -c builder concurrent --buffer build-dist-dev build-dist-prod
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Probably worth looking at transpiled output (webpack should be consuming `es/*` files). If it's an ES6 thing that _should_ be supported, we can look to updating the webpack build to use something more es-friendly...

**Comment by ryan-roemer** [comment]
And to save you round trips to CI, all of this should work/fail the same in localhost:

```sh
$ yarn run build-examples
$ yarn run lint
$ yarn run build
$ yarn run test-coverage
```

**Comment by dobesv** [comment]
Yeah, I'm working on the CI fixes.  I think I don't need to use `new.target`, which is what uglifyjs is mad about.  `this.constructor` should always be available and correct from my understanding.

**Comment by dobesv** [comment]
Looks like tests are passing.  Not sure if you have any other way to verify this won't break things for people who were using es native classes out there.  It does seem to fix the issue for me if I hand-patch node_modules using the generated file from `lib`.

**Comment by ryan-roemer** [comment]
@dobesv --  Great work! You changes still pass the original regression test I wrote for ES native classes at: https://github.com/FormidableLabs/radium/blob/0f7521fd1e5351ed203a5800a1c0a04bec0b9643/test/enhancer-test.js#L56-L66 so I _think_ we should be good.

@boygirl @alexlande -- We've got some great work here that I don't have time right now to kick the tires on. Can you recruit someone to vet this and possibly merge + release? Thanks!!!

**Comment by dobesv** [comment]
Any ETA on this one?

**Comment by ryan-roemer** [comment]
I believe @kylecesmat is going to look at this when he has a chance, hopefully this upcoming week. Thanks for your patience!

(I'd jump in and try to wrangle it, but I'm on parental leave with a newborn 👶 )

File: test/enhancer-test.js [hunk 1]
```diff
@@ -64,4 +64,66 @@ describe('Enhancer', () => {
...
+        ]);
+      }
+    }
+
+    const cleanup = require('jsdom-global')();
+    try {
+      const Enhanced = Enhancer(TestComponent);
+      const elt = React.createElement(Enhanced, {flag: true});
+      const {configure, mount} = require('enzyme');
+      const Adapter = require('enzyme-adapter-react-16');
```
Comment (kylecesmat) [inline test/enhancer-test.js]: I do want to see if we can refactor some of our existing tests to use jsdom/enzyme - in which case moving these to top-level requires. But this implementation is great for now.

File: .travis.yml [hunk 2]
```diff
@@ -6,9 +6,9 @@ addons:
   chrome: stable
 
 node_js:
-  - "4"
   - "6"
   - "8"
+  - "10"
```
Comment (kylecesmat) [inline .travis.yml]: Thanks for updating this!

File: src/enhancer.js [hunk 3]
```diff
@@ -92,27 +66,26 @@ export default function enhanceWithRadium(
...
-        //
-        // Instead, we just call `new` directly without the `_possibleConstructorReturn` wrapper.
-        const source = new OrigComponent(...arguments);
-
-        // Then we manually update context with properties.
-        copyProperties(source, this);
-
-        return this;
+        // Use Reflect.construct to simulate 'new'
+        const obj = Reflect.construct(
```
Comment (kylecesmat) [inline src/enhancer.js]: What are the implications of using `Reflect` in regards to cross-browser support? I believe IE11 is the only browser that will need a polyfill, would you be able to add a note in the README about including a polyfill like https://github.com/zloirock/core-js#ecmascript-reflect?

A note in this section would probably be best - https://github.com/FormidableLabs/radium#importing-radium

**Comment by kylecesmat** [comment]
This has been released in `radium@0.25.0`

File: src/enhancer.js (hunk 3 above)
Comment (dobesv) [reply src/enhancer.js]: @kylecesmat I don't think you can create a "native class" in IE11, so this code would never run in that browser.

File: test/enhancer-test.js (hunk 1 above)
Comment (dobesv) [reply test/enhancer-test.js]: Yeah, since I'm not a core contributor I wanted to have minimal impact.

File: .travis.yml (hunk 2 above)
Comment (dobesv) [reply .travis.yml]: Well, I had to  in order to get a green light on the PR :)

**Pull request merged**